workflows/analyze-package.yml

```diff
@@ -0,0 +1,81 @@
+name: Check analyzers
+description: >
+  This workflow checks the code quality of a Python package using various
+  analyzers like linters and type checkers, including ni-python-styleguide,
+  mypy, bandit, and pyright. It is designed to be reusable across different
+  projects and can be easily integrated into existing CI/CD pipelines.
+
+on:
+  workflow_call:
+    inputs:
+      project-directory:
+        description: Path to the directory containing pyproject.toml.
+        default: ${{ github.workspace }}
+      install-extras:
+        description: 'List of Poetry extras to install (comma separated)'
+        default: ''
+        required: false
+        type: string
+
+jobs:
+  check_analyzers:
+    name: Check analyzers for ${{ inputs.package-name }}
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        # Set the working-directory for all steps in this job.
+        working-directory: ${{ inputs.project-directory }}
+    steps:
+      - name: Get package name and version
+        id: get_package_info
+        run: |
+          result=$(poetry version -C "${{ inputs.project-directory }}")
+          name=$(echo "$result" | awk '{print $1}')
+          version=$(echo "$result" | awk '{print $2}')
+          echo "name=$name" >> $GITHUB_OUTPUT
+          echo "version=$version" >> $GITHUB_OUTPUT
+      - name: Check out repo
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          submodules: true
+      - name: Set up Python
+        uses: ./setup-python
+        id: setup-python
+      - name: Set up Poetry
+        uses: ./setup-poetry
+      - name: Check for lock changes
+        run: poetry check --lock
+      - name: Cache virtualenv
+        uses: actions/cache@0400d5f644dc74513175e3cd8d07132dd4860809 # v4.2.4
+        with:
+          path: ${{ inputs.project-directory }}/.venv
+          key: ${{ inputs.package-name }}-${{ runner.os }}-py${{ steps.setup-python.outputs.python-version }}-${{ hashFiles(format('{0}/poetry.lock', inputs.project-directory)) }}
+      - name: Install ${{ steps.get_package_info.outputs.name }}
+        run: |
+          if [ "${{ inputs.install-extras }}" != "" ]; then
+            poetry install -v --extras "${{ inputs.install-extras }}"
+          else
+            poetry install -v
+          fi
+      - name: Lint
+        run: poetry run ni-python-styleguide lint
+      - name: Mypy static analysis (Linux)
+        run: poetry run mypy
+      - name: Mypy static analysis (Windows)
+        run: poetry run mypy --platform win32
+      - name: Bandit security checks
+        run: poetry run bandit -c pyproject.toml -r src/
+      - name: Add virtualenv to the path for pyright-action
+        run: echo "$(poetry env info --path)/bin" >> $GITHUB_PATH
+      - name: Pyright static analysis (Linux)
```

Comment:
Pyright is technically "new", anything that is "new" should definitely go through a phase of being optional. Maybe this should follow what our internal pipeline does and check if the pyproject.toml lists pyright as a dep?